assets/src/edit-story/elements/image/display.js

```diff
@@ -47,12 +69,28 @@ function ImageDisplay({ element, box }) {
     focalY
   );
 
+  useEffect(() => {
+    let timeout;
+    if (window.webStoriesEditorResourceList[resource.id]?.type !== 'fullsize') {
+      timeout = setTimeout(async () => {
+        await preloadImage(resource.src);
+        window.webStoriesEditorResourceList[resource.id] = {
+          type: 'fullsize',
+        };
```

Comment:
You don't need to update the cached url here?

Reply:
We just need to read the fullsize type, then we get src from the resource:

```jsx
  if (resourceList[resource.id]?.type === 'fullsize' || resource.local) {
    initialSrcType = 'fullsize';
    initialSrc = resource.src;
  }
```